### PR TITLE
AbstractProfileService::convertAttributesToProfile sets profile id with username instead of id

### PR DESCRIPTION
**Pull request by marcadella**
Set profile id with id instead of username in legacy mode

### CODE REVIEW

File: pac4j-core/src/main/java/org/pac4j/core/profile/service/AbstractProfileService.java [hunk 1]
```diff
@@ -230,9 +230,15 @@ protected U convertAttributesToProfile(final List<Map<String, Object>> listStora
...
-                profile.setId(ProfileHelper.sanitizeIdentifier(profile, retrievedUsername));
+                profile.addAttribute(USERNAME, retrievedUsername);
             } else {
-                profile.setId(username);
+                profile.addAttribute(USERNAME, username);
+            }
+            final Object retrievedId = storageAttributes.get(getIdAttribute());
+            if (retrievedId != null) {
+                profile.setId(ProfileHelper.sanitizeIdentifier(profile, retrievedId));
+            } else {
```
Comment (mmoayyed) [inline pac4j-core/src/main/java/org/pac4j/core/profile/service/AbstractProfileService.java]: Else clause unnecessary.

Comment (marcadella) [reply pac4j-core/src/main/java/org/pac4j/core/profile/service/AbstractProfileService.java]: Well, most Technical exceptions are stuff that "should" not happen and are sanity check. But I can remove it.

**Comment by leleuj** [comment]
Can you elaborate on this?

**Comment by marcadella** [comment]
In the base code, `profile.setId(ProfileHelper.sanitizeIdentifier(profile, retrievedUsername));` sets the id with the username which is wrong in general: there is no reason why the id should equal the username. So I replaced this line with:
- `profile.setId(ProfileHelper.sanitizeIdentifier(profile, retrievedId));` to properly set the id
- `profile.addAttribute(USERNAME, retrievedUsername);` to set the username

**Comment by leleuj** [comment]
@marcadella Yes, this is strange, but it's the way the "legacy" mode works. Do you encounter any specific issue?

**Comment by marcadella** [comment]
Yes, on my setup id != username so the profile has an empty username and
the id is the username, which obviously is not right. I quick fixed it by
overriding the method but it would be nice if it was not needed.

ons. 14. aug. 2019 kl. 07:47 skrev LELEU Jérôme <notifications@github.com>:

**Comment by leleuj** [comment]
I don't consider this a bug, it's the way the legacy mode works and I'm reluctant to change that.

**Pull request closed without merging**